k8s/upgrade-job/src/upgrade/data_plane.rs

```diff
@@ -68,46 +85,82 @@ pub(crate) async fn upgrade_data_plane(namespace: String, rest_endpoint: String)
         );
 
         // Issue node drain command
-        issue_node_drain(node_name, &rest_client).await?;
-
-        // Wait for node drain to complete across the cluster.
-        wait_node_drain(&rest_client).await?;
+        drain_storage_node(node_name, &rest_client).await?;
 
         // Wait for any rebuild to complete.
         wait_for_rebuild(&rest_client).await?;
 
         // restart the data plane pod
-        restart_data_plane(node_name, pod, &k8s_client).await?;
+        delete_data_plane_pod(node_name, pod, &k8s_client).await?;
 
         // Uncordon the drained node
         uncordon_node(node_name, &rest_client).await?;
 
         // validate the new pod is up and running
-        verify_data_plane_pod_is_running(node_name, namespace.clone(), &k8s_client).await?;
+        verify_data_plane_pod_is_running(
+            node_name,
+            namespace.clone(),
+            &upgrade_to_version,
+            &k8s_client,
+        )
+        .await?;
 
         // Validate the control plane pod is up and running
-        is_control_plane_running(namespace.clone(), &k8s_client).await?;
+        verify_control_plane_is_running(namespace.clone(), &k8s_client, &upgrade_to_version)
+            .await?;
     }
     Ok(())
 }
 
 /// Uncordon storage Node.
-async fn uncordon_node(node_name: &str, rest_client: &RestClientSet) -> Result<()> {
-    rest_client
+async fn uncordon_node(node_id: &str, rest_client: &RestClientSet) -> Result<()> {
+    let storage_node = rest_client
         .nodes_api()
-        .delete_node_cordon(node_name, DRAIN_FOR_UPGRADE)
+        .get_node(node_id)
         .await
-        .context(StorageNodeUncordon {
-            node_name: node_name.to_string(),
+        .context(GetStorageNode {
+            node_id: node_id.to_string(),
         })?;
 
-    tracing::info!(node.name = node_name, "{PRODUCT} Node is uncordoned");
+    let drain_label_for_upgrade: String = DRAIN_FOR_UPGRADE.to_string();
 
-    Ok(())
+    loop {
+        match storage_node
```

Comment:
hmm `storage_node` never changes right so this loop will never end if it's Some(_) ?

Reply:
Did this a couple of places, I'll fix it.